src/Manager.js

```diff
@@ -0,0 +1,286 @@
+const { EventEmitter } = require("events");
+const merge = require("deepmerge");
+const { writeFile, readFile, exists } = require("fs");
+const { promisify } = require("util");
+const writeFileAsync = promisify(writeFile);
+const existsAsync = promisify(exists);
+const readFileAsync = promisify(readFile);
+const Config = require("./Config.js");
+const User = require("./User.js");
+const { defaultManagerOptions, defaultConfig } = require("./Constants.js");
+class VoiceManager extends EventEmitter {
+  constructor(client, options, init = true) {
+    super();
+    if (!client) throw new Error("Client is a required option.");
+    this.client = client;
+    this.ready = false;
+    this.users = [];
+    this.configs = [];
+    this.options = merge(defaultManagerOptions, options);
+    if (init) this._init();
+  }
+
+ updateConfig(guildID, updatedConfig) {
+    return new Promise(async (resolve, reject) => {
+      updatedConfig =
+        updatedConfig && typeof updatedConfig === "object"
+          ? merge(defaultConfig, updatedConfig)
+          : defaultConfig;
+      if (!guildID) {
+        return reject(`guildID is not valid. (val=${guildID})`);
+      }
+    });
+  }
+
+  async deleteUser(userID, guildID) {
+    await writeFileAsync(
+      this.options.userStorage,
+      JSON.stringify(this.users.map((user) => user.data)),
+      "utf-8"
+    );
+    this.refreshUserStorage();
+    return;
+  }
+
+  async deleteConfig(guildID) {
+    await writeFileAsync(
+      this.options.configStorage,
+      JSON.stringify(this.configs.map((config) => config.data)),
+      "utf-8"
+    );
+    this.refreshConfigStorage();
+    return;
+  }
+
+  async refreshUserStorage() {
+    return true;
+  }
+
+  async refreshConfigStorage() {
+    return true;
+  }
+
+  async editUser(_userID, _guildID, _userData) {
+    await writeFileAsync(
+      this.options.userStorage,
+      JSON.stringify(this.users.map((user) => user.data)),
+      "utf-8"
+    );
+    this.refreshUserStorage();
+    return;
+  }
+
+  async editConfig(_guildID, _configData) {
+    await writeFileAsync(
+      this.options.storage,
+      JSON.stringify(this.configs.map((config) => config.data)),
+      "utf-8"
+    );
+    this.refreshConfigStorage();
+    return;
+  }
+
+  async saveUser(userID, guildID, userData) {
+    await writeFileAsync(
+      this.options.userStorage,
+      JSON.stringify(this.users.map((user) => user.data)),
+      "utf-8"
+    );
+    this.refreshUserStorage();
+    return;
+  }
+
+  async saveConfig(guildID, configData) {
+    await writeFileAsync(
+      this.options.configStorage,
+      JSON.stringify(this.configs.map((config) => config.data)),
+      "utf-8"
+    );
+    this.refreshConfigStorage();
+    return;
+  }
```

Comment:
How does this work? I am a bit confused, you are not using any of the params...

Reply:
So basically the params will be used for the custom db's the user's provide for the default json db we used the configs and users array.

Reply:
Uh ok, but why do you need them

Reply:
To save the data.